frontend/src/components/Login/Login.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+
+const Login = () => {
+  const history = useHistory();
+
+  const tryLogin = async e => {
+    // Form validation handled by html5 <form> validation
+    e.preventDefault();
+    const res = await fetch('api/login', {
```

Comment:
kudos for trying to use native APIs (fetch). The only issue with it is that 
> the Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

You can add [missing error handling yourself](https://github.com/github/fetch#handling-http-error-statuses) or use axios or similar library instead.

Reply:
also consider creating some wrappers for get|post requests to incapsulate and reuse the code that initializes the HTTP request. 

```ts
api.get("/xxx")
api.post('/xxx', json)
```